app/mainwindow2.cpp

```diff
@@ -617,20 +633,21 @@ bool MainWindow2::saveDocument()
         return saveAsNewDocument();
 }
 
-bool MainWindow2::maybeSave()
+int MainWindow2::maybeSave()
 {
     if (mEditor->currentBackup() != mBackupAtSave)
     {
         int ret = QMessageBox::warning(this, tr("Warning"),
                                        tr("This animation has been modified.\n Do you want to save your changes?"),
-                                       QMessageBox::Save | QMessageBox::Discard | QMessageBox::Cancel,
-                                       QMessageBox::Save);
+                                       QMessageBox::Discard | QMessageBox::Save | QMessageBox::Cancel);
         if (ret == QMessageBox::Save)
-            return saveDocument();
+            return QMessageBox::Save ? saveDocument() : QMessageBox::Ignore;
```

Comment:
oh yes you're right about that... why did I add a ternary operator there anyway.. gimme a sec I'll fix it

Reply:
wait.. why will it always return true? we ask saveDocument returns true, then return QMessage::Save otherwise return QMessage::Ignore. I don't see how it always return true?

Reply:
I mean this line
`return QMessageBox::Save ? saveDocument() : QMessageBox::Ignore;`

Since the condition is `QMessageBox::Save`, which is not a zero value, so this ternary operator will always run `saveDocument()`. will never go to `QMessageBox::Ignore`

Reply:
Are you thinking in python? XDD

Reply:
Ohh right.. yes my bad I turned around the ternary... oh god XD too much Java coding lately perhaps...

Reply:
Should be fixed now P: